gui/account_events.go

```diff
@@ -82,12 +82,31 @@ func (u *gtkUI) handleMessageEvent(ev events.Message) {
 	timestamp := ev.When
 	encrypted := ev.Encrypted
 	message := ev.Body
+	fromSubscribedPeer := false
+
+	sender := ev.From.NoResource().String()
 
 	p, ok := u.getPeer(account, ev.From.NoResource())
 	if ok {
+		if p.Subscription != "" {
+			fromSubscribedPeer = true
+		}
 		p.LastSeen(ev.From)
 	}
 
+	if !fromSubscribedPeer && u.settings.GetIgnoreNonRoster() {
+		aa := account.session.GetConfig()
+		if aa.IgnoredSenders == nil {
+			// I couldn't figure out all construction paths of
+			// "Account" instances, so we'll init IgnoredSenders
+			// on demand
+			aa.IgnoredSenders = make(map[string]bool)
+		}
+		aa.IgnoredSenders[sender] = true
```

Comment:
Why do we keep track of this? It seems like a good way for a spammer to do a DoS attack on your config file, by just sending millions of messages from different automatically generated accounts?